### PR TITLE
better execute

### PR DESCRIPTION
**Pull request by TomasTomecek**
There still more room for improvements. The current implementation is heavily influenced by docker.

TODO:
 * [x] more tests

### CODE REVIEW

**Comment by TomasTomecek** [comment]
still not ideal; for non-blocking use case, there is no way to figure out that the command failed; the similar thing applies for interactive sessions

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/sAvG](https://da.gd/sAvG)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/7yFb](https://da.gd/7yFb)__
* Comment: __See details for logs__

File: conu/backend/docker/container.py [hunk 1]
```diff
@@ -249,28 +249,54 @@ def start(self):
...
+        This is a blocking call by default and writes output of the command to logger
+        using the INFO level -- this behavior can be changed if you set
+        the argument `blocking` to `False`.
+
+        If not blocking, you should consume the returned iterator in order to see logs or know
+        when the command finished:
+
+        ::
+
+            for line in self.execute(["ping", "8.8.8.8"])
```
Comment (dhodovsk) [inline conu/backend/docker/container.py]: Missing colon and 'blocking=False' argument.

Comment (TomasTomecek) [reply conu/backend/docker/container.py]: nice catch! I fixed even more

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/OdVm](https://da.gd/OdVm)__
* Comment: __See details for logs__

**Pull request merged**